pkg/kubernetes-common/credentialprovider/config_test.go

```diff
@@ -0,0 +1,266 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package credentialprovider
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+)
+
+func TestReadDockerConfigFile(t *testing.T) {
+	configJsonFileName := "config.json"
+	var fileInfo *os.File
+	preferredPaths := []string{}
+
+	//test dockerconfig json
+	inputDockerconfigJsonFile := "{ \"auths\": { \"http://foo.example.com\":{\"auth\":\"Zm9vOmJhcgo=\",\"email\":\"foo@example.com\"}}}"
+
+	preferredPath, err := ioutil.TempDir("", "test_foo_bar_dockerconfigjson_")
+	if err != nil {
+		t.Fatalf("Creating tmp dir fail: %v", err)
+		return
+	}
+	defer os.RemoveAll(preferredPath)
+	preferredPaths = append(preferredPaths, preferredPath)
```

Comment:
`pkg/kubernetes-common/credentialprovider/config_test.go:42:2:warning: ineffectual assignment to preferredPaths (ineffassign)`

Unused variable.

Reply:
this came straight from k8s code.  I guess i'll remove it but i have mixed feelings (i'd sort of like to keep the copied code as aligned w/ the original source as possible).